src/freertos_drivers/ti/CC32xxUart.cxx

```diff
@@ -119,6 +121,30 @@ void CC32xxUart::disable()
     MAP_UARTDisable(base);
 }
 
+/** Request an ioctl transaction
+ * @param file file reference for this device
+ * @param key ioctl key
+ * @param data key data
+ * @return 0 upon success, -errno upon failure
+ */
+int CC32xxUart::ioctl(File *file, unsigned long int key, unsigned long data)
+{
+    switch (key)
+    {
+        default:
+            return -EINVAL;
+        case TCSBRK:
```

Comment:
Use the IO* macros to decode this key for correctness.

Reply:
I'm not sure what you mean. I'm looking at the linux kernel and they use a case for this purpose like in the code here:
https://github.com/torvalds/linux/blob/d8a332730e757129e70675679f2b2a03f1ecf65e/drivers/tty/tty_io.c#L2539

Reply:
per offline conversation I'm marking this as done.